### ISSUE TITLE
Is .align() tractable with CMU-MOSEI?

### ISSUE THREAD

**Issue by tidd99** [A2Zadeh/CMU-MultimodalSDK#138]
I've been trying to align the CMU-MOSEI dataset to sentiment and emotion labels (separately). The initial alignment to word vectors was relatively fast, but I've left .align() running on the sentiment labels for two full days without the process completing. Is this normal? Does the SDK allow for aligning batches of data separately so that the entire set isn't loaded into memory?

Any help appreciated :-)

**Comment by A2Zadeh** [comment]
@tidd99 Good point. I am glad that the first step is fast. The second step should roughly take around 4 hours. I am not sure why it is taking such a long time. Is it align to words with collapse and then align to sentiment?

**Comment by tidd99** [comment]
Thanks for your reply! Yes, I've aligned to the words using your myavg() function and am now trying to align to sentiment. Roughly how much RAM would I need for it to complete within the 4 hours do you think?

**Comment by A2Zadeh** [comment]
@tidd99 I am thinking maybe around 32GB is needed. If you don't have that much, you can send me your code and I will run it for you :)

**Comment by tidd99** [comment]
A very kind offer, but I'm already working on code to do it in batches, so hopefully that works out. I just wanted to know if this option was already available with the SDK :)

**Comment by A2Zadeh** [comment]
@tidd99 great. Good luck to you with your experiments :). Please don't hesitate to contact me with any questions you might have. Here are some guidelines to get best performance out of the models:

1. myavg is a bit simple. You might consider weighted averaging based on intervals. 
2. definitely run your model with many hyperparameters on the validation set (~100 minimum). Multimodal data is a bit tricky.
3. Make sure you normalize features and also do feature selection.
